### PR TITLE
Move custom types to Node

### PR DESCRIPTION
**Pull request by Herringway**
Some big changes:
- unknown tags are no longer errors (as per spec)
- native type conversion is handled by Node and is handled a bit differently
  - requires the type to define an Type.opCast(T: Node) for Type -> YAML
  - requires a Type.this(Node) for Type <- YAML
- ~~Using regexes to detect type is no longer supported (and was very awkward/slow before)~~
- URI decoding doesn't suck as much anymore
- Constructor/Representer won't be public anymore

I still have to clean up a bunch of unittests here. Some are entirely in the wrong place. Docs also need an update.
The Constructor/Representer changes are incomplete, so don't rely on them just yet.

### CODE REVIEW

**Comment by Herringway** [comment]
I may not have to deprecate the custom regexes after all.

File: source/dyaml/representer.d [hunk 1]
```diff
@@ -100,6 +100,174 @@ Node representData(const Node data, ScalarStyle defaultScalarStyle, CollectionSt
...
+                Node("a", "tag:yaml.org,2002:str"),
+                Node("b", "tag:yaml.org,2002:str"),
+                Node("c", "tag:yaml.org,2002:str")
+            ], "tag:yaml.org,2002:seq"));
+    }
+}
+
+@safe unittest
+{
+    import std.stdio;
```
Comment (BBasile) [inline source/dyaml/representer.d]: to remove

Comment (Herringway) [reply source/dyaml/representer.d]: shhh you saw nothing

**Comment by Herringway** [comment]
I think I'm satisfied with where this is now.

**Pull request merged**